trieste/acquisition/function.py

```diff
@@ -304,14 +297,13 @@ def prepare_acquisition_function(
         """
         :param dataset: Unused.
         :param model: The model over the specified ``dataset``.
-        :return: The negative of the lower confidence bound function. This function will raise
-            `ValueError` if ``beta`` is negative.
+        :return: The expected improvement function. This function will raise
+            :exc:`ValueError` or :exc:`~tf.errors.InvalidArgumentError` if used with a batch size
+            greater than one.
+        :raise ValueError: If ``beta`` is negative.
         """
-        return lambda at: self._acquisition_function(model, self._beta, at)
-
-    @staticmethod
-    def _acquisition_function(model: ProbabilisticModel, beta: float, at: TensorType) -> TensorType:
-        return -lower_confidence_bound(model, beta, at)
+        lcb = lower_confidence_bound(model, self._beta)
+        return lambda at: -lcb(at)
```

Comment:
is this line necessary?

Reply:
do you mean splitting it in two? if so yes: it avoids evaluating everything in `lower_confidence_bound` which isn't part of the acquisition function, such as validating beta